docs/reference/contributing/target/ResetReason.md

```diff
@@ -0,0 +1,38 @@
+### Reset Reason
+
+The Reset Reason API provides a platform-independent method of checking the cause of the last system reset. Implementing the Reset Reason API is not mandatory, but by implementing it, you can check for erroneous system resets at boot time.
+
+#### Assumptions
+
+##### Defined behavior
+
+- The function `hal_reset_reason_clear` clears the reset reason registers for the next system boot.
+- The function `hal_reset_reason_get` will get the reset reason currently in the reset reason registers. This may not necessarily be the value of the reset reason registers that were set at boot time if they have been cleared since.
+
+##### Undefined behavior
+
+- The function `hal_reset_reason_get` is not guaranteed to return the same value when called repeatedly. The user should store the value for later use instead of calling the function repeatedly.
+- The function `hal_reset_reason_clear` may not clear the reset reason register in time for the next system boot.
+
+##### Potential bugs
+
+- The contents of the targets reset reason register may be cleared by some subsystem before it first gets called. This returns a `RESET_REASON_UNKNOWN` value to the user. To avoid this, the user should call the reset reason API as early as possible at boot time.
```

Comment:
Query: Is it accurate to replace this bullet with "A subsystem may clear the contents of the target's reset reason register before you first call the register. This returns a `RESET_REASON_UNKNOWN` value. To avoid this, call the Reset Reason API as early as possible at boot time." for active voice, consistent capitalization and consistent person?

Reply:
yep